tethys_gizmos/views/gizmos/jobs_table.py

```diff
@@ -140,11 +170,15 @@ def get_log_content(request, job_id, key1, key2=None):
         )
 
 
-def update_row(request, job_id):
+@async_login_required
+async def update_row(request, job_id):
     data = reconstruct_post_dict(request)
     try:
-        job = TethysJob.objects.get_subclass(id=job_id)
-        status = job.status
+        job = TethysJob.objects.get_subclass(id=job_id, user=request.user)
```

Comment:
As you probably suspect, this may not work for async. This is a reminder to test it.

Reply:
This was refactored to an async safe function